include/functions.h

```diff
@@ -37,6 +37,11 @@ void compute_displacement(const double dt, const size_t size, const Vector *velo
  */
 void displace_particles(const size_t size, const Vector *displacements, Particle *particles);
 
+/**
+  * Changes the displacement if the new position would surpass the X or Y limit.
+  */
+void fix_displacements(const size_t size, Particle *particles, const int n);
```

Comment:
Three things.

1. Please, put first all the `const` _arguments_ and then all the **mutable** ones.

2. Touching the limits should also _"stop"_ the particles in the given limit. Thus, you would need the **velocities** too.

3. Add tests!